smac/epm/gaussian_process.py

```diff
@@ -101,49 +89,41 @@ def _train(self, X: np.ndarray, y: np.ndarray, do_optimize: bool=True):
             the default hyperparameters of the kernel are used.
         """
 
-        if self.normalize_input:
-            # Normalize input to be in [0, 1]
-            self.X, self.lower, self.upper = normalization.zero_one_normalization(X, self.lower, self.upper)
-        else:
-            self.X = X
-
-        if len(y.shape) > 1:
-            y = y.flatten()
-            if len(y) != len(X):
-                raise ValueError('Shape mismatch: %s vs %s' % (y.shape, X.shape))
-
-        if self.normalize_output:
-            # Normalize output to have zero mean and unit standard deviation
-            self.y, self.y_mean, self.y_std = normalization.zero_mean_unit_var_normalization(y)
-            if self.y_std == 0:
-                raise ValueError("Cannot normalize output. All targets have the same value")
-        else:
-            self.y = y
+        if self.normalize_y:
+            y = self._normalize_y(y)
 
-        # Use the empirical mean of the data as mean for the GP
-        self.mean = np.mean(self.y, axis=0)
-        self.gp = george.GP(self.kernel, mean=self.mean)
+        n_tries = 10
+        for i in range(n_tries):
+            try:
+                # Instantiate a GP for each hyperparameter configuration
+                self.gp = skopt.learning.gaussian_process.GaussianProcessRegressor(
```

Comment:
Does this take a rng? If yes, shouldn't we use it?

Reply:
I just added the rng.